File: src/main/java/com/hohltier/decompiler/transformers/ProcyonTransformer.java
package com.hohltier.decompiler.transformers;

import com.hohltier.decompiler.models.FileModel;
import com.strobel.assembler.metadata.ArrayTypeLoader;
import com.strobel.decompiler.Decompiler;
import com.strobel.decompiler.DecompilerSettings;
import com.strobel.decompiler.ITextOutput;
import com.strobel.decompiler.PlainTextOutput;
import org.jetbrains.annotations.NotNull;

public class ProcyonTransformer implements ITransformer {

    @Override
    public String transform(@NotNull FileModel fileModel) {
        DecompilerSettings settings = DecompilerSettings.javaDefaults();
        ArrayTypeLoader typeLoader = new ArrayTypeLoader(fileModel.getBytes());
        settings.setTypeLoader(typeLoader);
        ITextOutput output = new PlainTextOutput();
        Decompiler.decompile(typeLoader.getClassNameFromArray(), output, settings);
        return output.toString().trim();
    }

}

File: src/main/kotlin/moe/sota/decompiler/transformers/ProcyonTransformer.kt
package moe.sota.decompiler.transformers

import com.strobel.assembler.metadata.ArrayTypeLoader
import com.strobel.decompiler.Decompiler
import com.strobel.decompiler.DecompilerSettings
import com.strobel.decompiler.PlainTextOutput
import moe.sota.decompiler.models.FileModel

class ProcyonTransformer : ITransformer {
    override fun transform(fileModel: FileModel): String {
        val settings = DecompilerSettings.javaDefaults()
        val typeLoader = ArrayTypeLoader(fileModel.bytes)
        settings.typeLoader = typeLoader
        val output = PlainTextOutput()
        Decompiler.decompile(typeLoader.classNameFromArray, output, settings)
        return output.toString()
    }
}
